setup.py

```diff
@@ -0,0 +1,100 @@
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from setuptools import setup, find_packages
+
+PROJECT_NAME = "ankaios_sdk"
+
+
+def generate_protos():
+    """Generate python protobuf files from the proto files."""
+    from grpc_tools import protoc
+
+    protos_dir = f"{PROJECT_NAME}/_protos"
+    proto_files = ["ank_base.proto", "control_api.proto"]
+
+    for proto_file in proto_files:
+        proto_path = os.path.join(protos_dir, proto_file)
+        output_file = proto_path.replace('.proto', '_pb2.py')
+
+        if not os.path.exists(output_file) or os.path.getmtime(proto_path) > os.path.getmtime(output_file):
+            print(f"Compiling {proto_path}...")
+            command = [
+                'grpc_tools.protoc',
+                f'-I={protos_dir}',
+                f'--python_out={protos_dir}',
+                f'--grpc_python_out={protos_dir}',
+                proto_path
+            ]
+            if protoc.main(command) != 0:
+                raise Exception(f"Error: {proto_file} compilation failed")
+
+            # Fix the import path in the generated control_api_pb2
+            # https://github.com/protocolbuffers/protobuf/issues/1491#issuecomment-261914766
+            if "control_api" in proto_file:
+                with open(output_file, 'r') as file:
+                    filedata = file.read()
+                    newdata = filedata.replace(
+                        "import ank_base_pb2 as ank__base__pb2",
+                        "from . import ank_base_pb2 as ank__base__pb2")
+                with open(output_file, 'w') as file:
+                    file.write(newdata)
+
+
+setup(
+    name=PROJECT_NAME,
+    version="0.1.0",
+    license="Apache-2.0",
+    author="Elektrobit Automotive GmbH and Ankaios contributors",
+    # author_email="",
+    description="Eclipse Ankaios Python SDK - provides a convenient Python interface for interacting with the Ankaios platform.",
+    long_description=open('README.md').read(),
+    long_description_content_type="text/markdown",
+    url="https://eclipse-ankaios.github.io/ankaios/latest/",
+    python_requires='>=3.6',
+    package_dir={'': '.'},
+    packages=find_packages(where="."),
+    include_package_data=True,
+    classifiers=[
+        "Programming Language :: Python :: 3",
+        "License :: OSI Approved :: Apache Software License",
+        "Operating System :: OS Independent",
+    ],
+    project_urls={
+        "Documentation": "https://eclipse-ankaios.github.io/ankaios/latest/",
+        "Source": "https://github.com/eclipse-ankaios/ank-sdk-python",
+        "Bug Tracker": "https://github.com/eclipse-ankaios/ank-sdk-python/issues",
+    },
+    install_requires=[
+        "protobuf==5.27.2",
+        "PyYAML",
+    ],
+    setup_requires=[
+        "protobuf==5.27.2",
+        "grpcio-tools>=1.56.2",
```

Comment:
Yes. There are 2 approaches on using protoc:
- Install protoc and run the protoc command directly from a shell.
- Use protoc directly from the grpcio-tools python library.
The advantage of the second option is that it is platform independent. This will work on windows as well, for instance. The first option on windows would be a lot more troublesome.

Reply:
Maybe we can document this.

Reply:
Wrote a small comment in setup.py, where it is specified. I don't know of a better place to document it.

With this occasion, documented all the dependencies.